python/tchannel/tornado/dispatch.py

```diff
@@ -88,17 +93,25 @@ def handle_call(self, request, connection):
                 request.id,
             )
         else:
-            response = Response(id=request.id, connection=connection)
-            yield self._call_endpoint(endpoint, request, response)
-
-    def route(self, rule, **opts):
+            response = Response(id=request.id,
+                                tracing=request.tracing,
+                                connection=connection)
+
+            from .tchannel import TChannelProxy
```

Comment:
Yeah, if you need to do this, the class shouldn't be defined in the same class as TChannel.